Mu2eG4/inc/Mu2eG4Config.hh

```diff
@@ -139,6 +137,8 @@ namespace mu2e {
                   "or one of \"PDG\", \"All\", \"None\" pre-defined settings.")
           };
 
+      OptionalDelegatedParameter mscTransitionEnergy {Name("mscTransitionEnergy")};
```

Comment:
Is there a reason to use OptionalDelegatedParameter here?  It looks like mscTransitionEnergy can be a more explicit OptionalAtom<double>

Reply:
I was looking for a way to take an action only if the keyword was present; This was the best way I could find. 
Maybe I missed it, but what I wanted, was to have an equivalent of has_key in Mu2eG4Config::Physics , but could not find it.

Reply:
DelegatedParameter objects are for the cases when the FHICL content can not be represented as a C++ structure known at the compilation time.  For example our G4Cuts with their arbitrary nesting.    Or BirksConsts, where the keys in key:value pairs are arbitrary and can not be handled directly without either DelegatedParameter or modifying the legacy FHICL syntax.   In our case just declare mscTransitionEnergy as OptionalAtom of double.    There may be no exact equivalent for has_key, rather OptionalAtom behaves like get_if_present.   You will still need to declare

   double mscModelTransitionEnergy(std::numeric_limits<double>::max()); 

then the call

   phys.mscModelTransitionEnergy(mscModelTransitionEnergy)

returns a boolean: true means the key was present and the variable mscModelTransitionEnergy now has the value from FHICL.  False means it was not specified, the value is unchanged from before the call.  An example of OptionalAtom use is defaultMinPointDistance() in Mu2eG4/src/Mu2eG4TrajectoryControl.cc

Reply:
Thanks Andrei. Done.
This is actually documented, but I missed it the other day. I put the links here for a future reference:
https://cdcvs.fnal.gov/redmine/projects/fhicl-cpp/wiki/Configuration_validation_and_fhiclcpp_types#Optional-parameters
https://cdcvs.fnal.gov/redmine/projects/fhicl-cpp/wiki/Fhiclcpp_types_in_detail#Optional-parameters-argument-types